`PrivateSentrySDKOnly` isn't compiling for macOS

### Description

> 2023-08-09 15:30:12.805 xcodebuild[7721:2020644] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
> --- xcodebuild: WARNING: Using the first of multiple matching destinations:
> { platform:macOS, arch:arm64, id:00006021-000229912E47401E }
> { platform:macOS, arch:x86_64, id:00006021-000229912E47401E }
> /Users/user/Github/sentry-dart/flutter/ios/Classes/SentryFlutterPluginApple.swift:469:62: error: type 'PrivateSentrySDKOnly' has no member 'appStartMeasurement'
>         guard let appStartMeasurement = PrivateSentrySDKOnly.appStartMeasurement else {
>                                         ~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~
> @property (class, nullable, nonatomic, readonly) SentryAppStartMeasurement *appStartMeasurement;
> ^
> @property (class, nullable, nonatomic, readonly) SentryAppStartMeasurement *appStartMeasurement;
> ^
> @class SentryDebugMeta, SentryAppStartMeasurement, SentryScreenFrames, SentryOptions,
> ^
> /Users/user/Github/sentry-dart/flutter/ios/Classes/SentryFlutterPluginApple.swift:476:56: error: cannot infer contextual base in reference to member 'cold'
>         let isColdStart = appStartMeasurement.type == .cold
>                                                       ~^~~~
> warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Flutter Assemble' from project 'Runner')
> /Users/user/Github/sentry-dart/flutter/example/macos/Pods/Pods.xcodeproj: warning: The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.11, but the range of supported deployment target versions is 10.13 to 13.3.99. (in target 'FMDB' from project 'Pods')
> ** BUILD FAILED **

@brustolin @armcknight did something change for `PrivateSentrySDKOnly`? I've not found any changes but not sure why its not compiling for macOS anymore.

--- a/.github/workflows/flutter.yml
+++ b/.github/workflows/flutter.yml
@@ -160,7 +160,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       # https://github.com/CocoaPods/CocoaPods/issues/5275#issuecomment-315461879
-      - run: pod lib lint ios/sentry_flutter.podspec --configuration=Debug --skip-import-validation --allow-warnings
+      - run: pod lib lint ios/sentry_flutter.podspec --configuration=Debug --skip-import-validation --allow-warnings --verbose
 
   swift-lint:
     runs-on: ubuntu-latest

--- a/flutter/example/ios/Podfile
+++ b/flutter/example/ios/Podfile
@@ -1,7 +1,5 @@
-wanted_project_target = '11.0'
-
 # Uncomment this line to define a global platform for your project
-platform :ios, wanted_project_target
+platform :ios, '11.0'
 
 # CocoaPods analytics sends network stats synchronously affecting flutter build latency.
 ENV['COCOAPODS_DISABLE_STATS'] = 'true'
@@ -37,15 +35,6 @@ target 'Runner' do
 end
 
 post_install do |installer|
-  # remove after https://github.com/flutter/flutter/issues/124340 getting into all channels
-  installer.generated_projects.each do |project|
-    project.targets.each do |target|
-        target.build_configurations.each do |config|
-            config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = wanted_project_target
-        end
-    end
-  end
-
   installer.pods_project.targets.each do |target|
     flutter_additional_ios_build_settings(target)
   end

--- a/flutter/ios/Classes/SentryFlutterPluginApple.swift
+++ b/flutter/ios/Classes/SentryFlutterPluginApple.swift
@@ -466,6 +466,7 @@ public class SentryFlutterPluginApple: NSObject, FlutterPlugin {
     }
 
     private func fetchNativeAppStart(result: @escaping FlutterResult) {
+        #if os(iOS) || os(tvOS)
         guard let appStartMeasurement = PrivateSentrySDKOnly.appStartMeasurement else {
             print("warning: appStartMeasurement is null")
             result(nil)
@@ -481,6 +482,10 @@ public class SentryFlutterPluginApple: NSObject, FlutterPlugin {
         ]
 
         result(item)
+        #else
+            print("note: appStartMeasurement not available on this platform")
+            result(nil)
+        #endif
     }
 
     private var totalFrames: UInt = 0

--- a/flutter/ios/sentry_flutter.podspec
+++ b/flutter/ios/sentry_flutter.podspec
@@ -6,7 +6,7 @@ Pod::Spec.new do |s|
 Sentry SDK for Flutter with support to native through sentry-cocoa.
                        DESC
   s.homepage         = 'https://sentry.io'
-  s.license          = { :file => '../LICENSE' }
+  s.license          = { :type => 'MIT', :file => '../LICENSE' }
   s.authors          = "Sentry"
   s.source           = { :git => "https://github.com/getsentry/sentry-dart.git",
                          :tag => s.version.to_s }

